### ISSUE TITLE
Font Size

### ISSUE THREAD

**Issue by git42leech** [LocalIdentity/PathOfBuilding#218]
Scaling font size would be a cool feature for bigger screens.

**Comment by toddf2p** [comment]
Visually impaired here, bold font I think would help a lot. Also one of the red colored fonts is much easier to read than others.

Example: https://imgur.com/a/9xN8G2t

"Mines can be detonated an additional time" is far easier to read than the other red text below. I don't know if that's due to my particular visual impairment or my monitor configuration or if it's universally true for others as well.

Thank you.

**Comment by pHiney** [comment]
Need to see where the font is setup. I'll check tonight. I have a 4k screen and when i was looking at the Thousands and Decimal separators (#244) , i could not see the difference between the comma and the period. **Note though** that changing font size is going to cause issues with layout. It might lead to a nearly never ending set of layout issues.

**Comment by ppoelzl** [comment]
The layouting almost exclusivly uses absolute positioning and the SimpleGraphic library is not really made for building scaling UIs. Any requests for changes to window/font scaling or unicode support are essentially asking for a complete rewrite of the graphics frontend using some technology that supports all of that out of the box.

**Comment by luishdz1010** [comment]
What about setting a scaling factor for everything? Not just fonts. This should be easier right?

**Comment by LocalIdentity** [comment]
You are able to change the font size in the notes tab now with `Ctrl` + `Mouse wheel`
Adding font scaling to the rest of PoB is very difficult and state by ppoelzl above

**Comment by chx** [comment]
To add to the chorus and hoping maybe in 2024 there would be time to fix this, here's how it looks at 300% scaling on a 55" 4K TV: ![PXL_20231202_200728837](https://github.com/PathOfBuildingCommunity/PathOfBuilding/assets/193045/33a80a1c-2ab5-4a34-acfb-fc63ef75dd93). Path Of Pathing has the same issue but curiously, poeplanner looks beautiful.

**Comment by pHiney** [comment]
@chx Under what conditions does poeplanner look beautiful ?

I think PoB looks pretty good in your screen shot - what was it you wanted to point out ?

thanks
X

**Comment by chx** [comment]
Try setting Windows OS scaling to 300% and open PoB and https://pathofpathing.com/ and check them against https://poeplanner.com/ and https://vilsol.github.io/timeless-jewels/tree and I think you will immediately see what I mean. 

This is not a critique of PoB, I love you guys.

**Comment by chx** [comment]
I just found it curious that one website out of the three behaved the same. 

Anyways, everything you describe sounds familiar from a long, long time ago even if we didn't need to struggle with DPI but we did struggle when resolution kinda exploded on PCs. I remember GEM/Ventura looking absolutely terrible initially on SVGA (and yes, I am old). 

I wish I knew enough Lua to help :(

**Comment by Xeverous** [comment]
@zao I recently got informed that PoB has the x/y (and other UI math) hardcoded. I'm willing to contribute and have an idea.

Make an abstraction for UI coordination/positioning and move it to the Simple Graphic library. Let Lua code only handle the logic. Basically, when creating a button don't pass any x/y stuff and the layouting can be done dynamically on a simple V-Box/H-Box widget tree (just pure tree/graph math, no HTML/CSS involved).

I have a working demo that implements this layouting concept, written in C++ (I have no Lua skills) and demonstrates some additional features like scrolling, hiding, spacing control.

**Comment by he-dev** [comment]
Is there any chance this issue gets more attention soon?

**Comment by LocalIdentity** [comment]
@Xeverous we now include larger font sizes in the repo up to size 64. Most of the strings are size 14-20 by default so when you scale a size 20 font by 2 we just use the size 40 one so it's still crisp

**Comment by LocalIdentity** [comment]
@he-dev yeah we haven't released it yet as I've been fixing bugs with it over the past day or so. It'll be in the next release for both PoB 1 and 2
